src/protocol/basics/check_zero.rs

```diff
@@ -107,9 +107,21 @@ mod tests {
                 counter += 1;
 
                 let protocol_output = tokio::try_join!(
-                    check_zero(context[0].narrow(&iteration), record_id, &v_shares[0],),
-                    check_zero(context[1].narrow(&iteration), record_id, &v_shares[1],),
-                    check_zero(context[2].narrow(&iteration), record_id, &v_shares[2],),
+                    check_zero(
+                        context[0].narrow(&iteration).set_total_records(1),
```

Comment:
> Separately, I wonder if it would be easier all around if TestWorld::contexts() provided total records of 1 by default.

I had a similar thought of adding a record count argument to the `world.malicious` and `world.semi_honest` helpers (which could then pass it to `contexts()`).

Any thoughts on `async fn semi_honest(&self, total_records: Option<NonZeroUsize>, input: I, helper_fn: H) -> [O; 3]` and `fn contexts(&self, total_records: Option<NonZeroUsize>)`?

Reply:
Given how often we do something != 1, it might be easier to do the `contexts(&self, tr: Option<NonZeroSize>)` and `TestWorld::set_total_records()` so that we don't need a second signature for `semi_honest()`.  It's state, but only on `TestWorld`.  The others are also fine.